src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
Mhh, looks liek another weakness of the eclipse compiler. (Eclipse compiles it without problems)

An alternative solution:
````java
protected final Iterable<Text> childrenIterable = () ->
        getChildren().isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
````